README.adoc

```diff
@@ -0,0 +1,223 @@
+
+Originally, `git ls-files` inside the default gemspec template was designed so that users
+publishing their first gem wouldn't unintentionally publish artifacts to it. +
+Now that `bundler` won't let you release if you have uncommitted stuff in your working
+repository, the risk is quite lower (if none at all!).
+
+==== Examples [[gemspec-git-examples]]
+
+[source,ruby]
+----
+# bad
+Gem::Specification.new do |spec|
+  spec.files         = `git ls-files`.split('\n')
+  spec.test_files    = `git ls-files -- spec/*`.split('\n')
+end
+
+# good
+Gem::Specification.new do |spec|
+  spec.files         = Dir['lib/**/*', 'LICENSE', 'README.md']
+  spec.test_files    = Dir['spec/**/*']
+end
+
+# bad
+Gem::Specification.new do |spec|
+  spec.files         = Dir.chdir(File.expand_path('..', __FILE__)) do
+    `git ls-files -z`.split('\\x0').reject { |f| f.match(%r{^(test|spec|features)/}) }
+  end
+end
+
+# good
+Gem::Specification.new do |spec|
+  spec.files         = Rake::FileList['**/*'].exclude(*File.read('.gitignore').split)
+end
+
+# bad
+Gem::Specification.new do |spec|
+  spec.files         = `git ls-files`.split('\n')
+  spec.test_files    = `git ls-files -- test/{functional,unit}/*`.split('\n')
+  spec.executables   = `git ls-files -- bin/*`.split('\n').map { |f| File.basename(f) }
+end
+
+# good
+Gem::Specification.new do |spec|
+  spec.files         = Dir.glob('lib/**/*')
+  spec.test_files    = Dir.glob('test/{functional,test}/*')
+  spec.executables   = Dir.glob('bin/*').map { |f| File.basename(f) }
+end
+----
+
+
+=== Relative Require To Lib [[require-relative-to-lib]]
+
+About the `RelativeRequireToLib` cop:
+
+==== Rationale [[require-relative-to-lib-rationale]]
+
+Avoid using `require_relative` with relative path to lib. Use `require` instead.
+
+==== Need [[require-relative-to-lib-need]]
+
+As mentioned earlier, packages in Debian are built in a clean environment. After building them,
+we also run the tests with the `lib/` directory removed to force those tests to use the
+library installed globally via the Debian package. +
+We do so because we want to ensure that the library is loaded from the system location and not
+from the source tree and that it works like it is intended to.
+
+Therefore, when one uses relative path, we end up getting a `LoadError`, stating: +
+`cannot load such file -- /<<PKGBUILDDIR>>/foo`.
+
+Of course, that said, with this cop, we don't intend to disregard the usasge of `require_relative`
+in general. Not at all! It is still recommended but we just add one tiny exception to it. +
+That is, avoid using `require_relative` with relative path to `lib/` in your tests. That's
+it, that's all we ask!
+
+==== Examples [[require-relative-to-lib-examples]]
+
+[source,ruby]
+----
+# bad
+require_relative '../../lib/foo/bar'
+
+# good
+require 'foo/bar'
+
+# bad
+require_relative 'lib/foo.rb'
+
+# good
+require 'foo.rb'
+
+# good
+require_relative 'spec_helper'
+require_relative 'foo/bar'
```

Comment:
require 'foo/bar' I assume

Reply:
I wanted to express that `require_relative` is okay to use inside `lib/` or inside `specs/`, as long as it is not cross spec-lib.
So I'll perhaps amend it to `require_relative 'spec/foo/bar'.